### ISSUE TITLE
Failing on 10.13 due to missing fuse.pc

### ISSUE THREAD

**Issue by bram2000** [geerlingguy/ansible-role-homebrew#92]
I'm running your vanilla version of mac-dev-playbook with the default config (have tried my own config with the same result) and hitting this strange error in the homebrew role:

```
TASK [geerlingguy.homebrew : Ensure proper ownership on homebrew_install_path subdirs.] *************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OSError: [Errno 2] No such file or directory: '/usr/local/lib/pkgconfig/fuse.pc'
fatal: [127.0.0.1]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_zlVUfm/ansible_module_file.py\", line 473, in <module>\n    main()\n  File \"/tmp/ansible_zlVUfm/ansible_module_file.py\", line 350, in main\n    changed |= recursive_set_attributes(module, to_bytes(file_args['path'], errors='surrogate_or_strict'), follow, file_args)\n  File \"/tmp/ansible_zlVUfm/ansible_module_file.py\", line 172, in recursive_set_attributes\n    changed |= module.set_fs_attributes_if_different(tmp_file_args, changed, expand=False)\n  File \"/tmp/ansible_zlVUfm/ansible_modlib.zip/ansible/module_utils/basic.py\", line 1507, in set_fs_attributes_if_different\n  File \"/tmp/ansible_zlVUfm/ansible_modlib.zip/ansible/module_utils/basic.py\", line 1164, in set_owner_if_different\n  File \"/tmp/ansible_zlVUfm/ansible_modlib.zip/ansible/module_utils/basic.py\", line 1059, in user_and_group\nOSError: [Errno 2] No such file or directory: '/usr/local/lib/pkgconfig/fuse.pc'\n", "module_stdout": "", "msg": "MODULE FAILURE", "rc": 1}
    to retry, use: --limit @/Users/jbramley/Code/vanilla-playbook/main.retry
```

**Comment by bram2000** [comment]
Looks like this is also failing on the mac-dev-playbook Travis CI build. I wonder why/how it is working on the Travis build on this repo.

**Comment by bram2000** [comment]
Just tested with the example playbook from the README and seeing the same failure.

**Comment by geerlingguy** [comment]
See related: https://github.com/geerlingguy/ansible-role-mas/issues/12

**Comment by geerlingguy** [comment]
Latest CI tests are broken too: https://travis-ci.org/geerlingguy/ansible-role-homebrew/builds/373041096

I wonder what changed here... tests were passing locally and in Travis CI a week ago!

**Comment by geerlingguy** [comment]
Got a bunch of new errors in previous fix attempt:

```
TASK [ansible-role-homebrew : Ensure proper homebrew folders are in place] *****
changed: [localhost] => (item=Cellar)
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OSError: [Errno 2] No such file or directory: '/usr/local/lib/pkgconfig/fuse.pc'
failed: [localhost] (item=Homebrew) => {"changed": false, "item": "Homebrew", "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_heJqzA/ansible_module_file.py\", line 473, in <module>\n    main()\n  File \"/tmp/ansible_heJqzA/ansible_module_file.py\", line 350, in main\n    changed |= recursive_set_attributes(module, to_bytes(file_args['path'], errors='surrogate_or_strict'), follow, file_args)\n  File \"/tmp/ansible_heJqzA/ansible_module_file.py\", line 172, in recursive_set_attributes\n    changed |= module.set_fs_attributes_if_different(tmp_file_args, changed, expand=False)\n  File \"/tmp/ansible_heJqzA/ansible_modlib.zip/ansible/module_utils/basic.py\", line 1507, in set_fs_attributes_if_different\n  File \"/tmp/ansible_heJqzA/ansible_modlib.zip/ansible/module_utils/basic.py\", line 1164, in set_owner_if_different\n  File \"/tmp/ansible_heJqzA/ansible_modlib.zip/ansible/module_utils/basic.py\", line 1059, in user_and_group\nOSError: [Errno 2] No such file or directory: '/usr/local/lib/pkgconfig/fuse.pc'\n", "module_stdout": "", "msg": "MODULE FAILURE", "rc": 1}
changed: [localhost] => (item=Frameworks)
changed: [localhost] => (item=Caskroom)
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OSError: [Errno 2] No such file or directory: '/usr/local/Caskroom/oclint/0.11,15.6.0/oclint-0.11/bin/oclint'
failed: [localhost] (item=bin) => {"changed": false, "item": "bin", "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_NBbZPO/ansible_module_file.py\", line 473, in <module>\n    main()\n  File \"/tmp/ansible_NBbZPO/ansible_module_file.py\", line 350, in main\n    changed |= recursive_set_attributes(module, to_bytes(file_args['path'], errors='surrogate_or_strict'), follow, file_args)\n  File \"/tmp/ansible_NBbZPO/ansible_module_file.py\", line 172, in recursive_set_attributes\n    changed |= module.set_fs_attributes_if_different(tmp_file_args, changed, expand=False)\n  File \"/tmp/ansible_NBbZPO/ansible_modlib.zip/ansible/module_utils/basic.py\", line 1507, in set_fs_attributes_if_different\n  File \"/tmp/ansible_NBbZPO/ansible_modlib.zip/ansible/module_utils/basic.py\", line 1164, in set_owner_if_different\n  File \"/tmp/ansible_NBbZPO/ansible_modlib.zip/ansible/module_utils/basic.py\", line 1059, in user_and_group\nOSError: [Errno 2] No such file or directory: '/usr/local/Caskroom/oclint/0.11,15.6.0/oclint-0.11/bin/oclint'\n", "module_stdout": "", "msg": "MODULE FAILURE", "rc": 1}
ok: [localhost] => (item=etc)
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OSError: [Errno 2] No such file or directory: '/usr/local/Caskroom/oclint/0.11,15.6.0/oclint-0.11/include/c++'
failed: [localhost] (item=include) => {"changed": false, "item": "include", "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_GpG2Hq/ansible_module_file.py\", line 473, in <module>\n    main()\n  File \"/tmp/ansible_GpG2Hq/ansible_module_file.py\", line 350, in main\n    changed |= recursive_set_attributes(module, to_bytes(file_args['path'], errors='surrogate_or_strict'), follow, file_args)\n  File \"/tmp/ansible_GpG2Hq/ansible_module_file.py\", line 172, in recursive_set_attributes\n    changed |= module.set_fs_attributes_if_different(tmp_file_args, changed, expand=False)\n  File \"/tmp/ansible_GpG2Hq/ansible_modlib.zip/ansible/module_utils/basic.py\", line 1507, in set_fs_attributes_if_different\n  File \"/tmp/ansible_GpG2Hq/ansible_modlib.zip/ansible/module_utils/basic.py\", line 1164, in set_owner_if_different\n  File \"/tmp/ansible_GpG2Hq/ansible_modlib.zip/ansible/module_utils/basic.py\", line 1059, in user_and_group\nOSError: [Errno 2] No such file or directory: '/usr/local/Caskroom/oclint/0.11,15.6.0/oclint-0.11/include/c++'\n", "module_stdout": "", "msg": "MODULE FAILURE", "rc": 1}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OSError: [Errno 2] No such file or directory: '/usr/local/Caskroom/oclint/0.11,15.6.0/oclint-0.11/lib/oclint'
failed: [localhost] (item=lib) => {"changed": false, "item": "lib", "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_xRhhdr/ansible_module_file.py\", line 473, in <module>\n    main()\n  File \"/tmp/ansible_xRhhdr/ansible_module_file.py\", line 350, in main\n    changed |= recursive_set_attributes(module, to_bytes(file_args['path'], errors='surrogate_or_strict'), follow, file_args)\n  File \"/tmp/ansible_xRhhdr/ansible_module_file.py\", line 172, in recursive_set_attributes\n    changed |= module.set_fs_attributes_if_different(tmp_file_args, changed, expand=False)\n  File \"/tmp/ansible_xRhhdr/ansible_modlib.zip/ansible/module_utils/basic.py\", line 1507, in set_fs_attributes_if_different\n  File \"/tmp/ansible_xRhhdr/ansible_modlib.zip/ansible/module_utils/basic.py\", line 1164, in set_owner_if_different\n  File \"/tmp/ansible_xRhhdr/ansible_modlib.zip/ansible/module_utils/basic.py\", line 1059, in user_and_group\nOSError: [Errno 2] No such file or directory: '/usr/local/Caskroom/oclint/0.11,15.6.0/oclint-0.11/lib/oclint'\n", "module_stdout": "", "msg": "MODULE FAILURE", "rc": 1}
changed: [localhost] => (item=opt)
changed: [localhost] => (item=sbin)
changed: [localhost] => (item=share)
changed: [localhost] => (item=share/zsh)
changed: [localhost] => (item=share/zsh/site-functions)
ok: [localhost] => (item=var)
```

Looks like the `oclint` cask screws some things up too. Trying to remove the recurse flag on that task too and will see what that does...

**Comment by geerlingguy** [comment]
Fixed via commits above—I just tagged and released 2.4.1, which has the removal of the 'recurse' option that was causing these issues. Still not 100% sure why they crept in, but the role and homebrew itself should still work okay without recursively modifying the attributes—I believe that bit was added during a transitional phase in Homebrew's major release a year or two ago, and shouldn't be needed for new installs.

**Comment by geerlingguy** [comment]
@elliotweiser - Maybe. Does that mess with idempotence though?
